src/main/java/com/epam/esm/controller/GiftCertificatesController.java

```diff
@@ -100,12 +108,12 @@ public GiftCertificatesController(GiftCertificateService service) {
      * @param id - id of {@link GiftCertificate} object to be updated.
      * @param giftCertificate - new data for updating an {@link GiftCertificate} object to be updated.
      * @return {@link ResponseEntity} contained both {@link HttpStatus} status and updated {@link GiftCertificate} object.
-     * @throws BadEntityException in case if invalid data passed in request to upadte a {@link GiftCertificate} object.
-     * @throws EntityNotExistsException in case if nothing found with searched id.
      */
     @PatchMapping(value = "/{id}", produces = JSON)
-    public ResponseEntity<GiftCertificateDto> updateGiftCertificate(@PathVariable(ID) long id, @RequestBody GiftCertificateDto giftCertificate) throws EntityNotExistsException, BadEntityException {
-        return new ResponseEntity<>(service.updateGiftCertificate(id, giftCertificate), HttpStatus.OK);
+    public ResponseEntity<GiftCertificateDto> updateGiftCertificate(@PathVariable(ID) long id, @RequestBody GiftCertificateDto giftCertificate) {
+        GiftCertificateDto updatedGiftCertificate = service.updateGiftCertificate(id, giftCertificate);
+        updatedGiftCertificate.add(linkTo(methodOn(GiftCertificatesController.class).getGiftCertificate(updatedGiftCertificate.getId())).withSelfRel());
```

Comment:
This part `linkTo(methodOn(GiftCertificatesController.class).getGiftCertificate(updatedGiftCertificate.getId())).withSelfRel()` is duplicated everywhere in the class. Please fix this